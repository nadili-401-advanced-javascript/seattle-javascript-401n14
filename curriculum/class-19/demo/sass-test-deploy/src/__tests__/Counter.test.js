import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Counter from '../components/Counter';

configure({ adapter: new Adapter() });

describe('Counter component', () => {
    it('renders correctly', () => {
        const component = renderer.create(<Counter count={-1} />).toJSON();
        expect(component).toMatchSnapshot();
    });

    it('correctly adds one to state count variable', () => {
        let component = shallow(<Counter />);
        let addBtn = component.find('div.bordered-box button#add');

        addBtn.simulate('click');
        expect(component.state('count')).toBe(1);
    });

    it('correctly adds one to state count variable', () => {
        let component = render(<Counter />);
        expect(component.find('div.bordered-box h2')).toBeDefined();
    });
});
