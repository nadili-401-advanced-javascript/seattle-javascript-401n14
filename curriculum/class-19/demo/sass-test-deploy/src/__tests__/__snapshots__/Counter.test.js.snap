// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component renders correctly 1`] = `
<div
  className="bordered-box"
>
  <h2>
    Your current count is: 
    <span>
      -1
    </span>
  </h2>
  <button
    id="subtract"
    onClick={[Function]}
  >
    Subtract
  </button>
  <button
    id="add"
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;
